src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -90,68 +78,54 @@ public void InvokeCompletionCallback(uint sniErrorCode)
         /// Allocate space for data
         /// </summary>
         /// <param name="capacity">Length of byte array to be allocated</param>
-        public void Allocate(int capacity)
+        public void Allocate(int capacity, bool reserveMuxHeader)
         {
-            if (_data != null && _data.Length < capacity)
+            SNIPacketFlags flags = reserveMuxHeader ? SNIPacketFlags.MuxHeaderReserved : SNIPacketFlags.None;
+            int headerCapacity = reserveMuxHeader ? SNISMUXHeader.HEADER_LENGTH : 0;
+            int totalCapacity = headerCapacity + capacity;
+            if (_data != null)
```

Comment:
_data will always be null at this point, since Allocate is called from the constr. This path of checking if _Data is null or not, is not needed.

Reply:
Simplified the entire file. Allocate could be moved inline in the ctor, worth doing?